solidity/contracts/deposit/Deposit.sol

```diff
@@ -385,89 +411,230 @@ contract Deposit is DepositFactoryAuthority {
+    /// @param _tbtcToken `TBTCToken` contract. More info in TBTCToken`.
+    /// @param _tbtcDepositToken `TBTCDepositToken` (TDT) contract. More info in
+    ///        `TBTCDepositToken`.
+    /// @param _feeRebateToken `FeeRebateToken` (FRT) contract. More info in
+    ///        `FeeRebateToken`.
+    /// @param _vendingMachineAddress `VendingMachine` address. More info in
+    ///        `VendingMachine`.
+    /// @param _m Signing group honesty threshold.
+    /// @param _n Signing group size.
+    /// @param _lotSizeSatoshis The minimum amount of satoshi the funder is
+    ///                         required to send. This is also the amount of
+    ///                         TBTC the TDT holder will be eligible to mint:
+    ///                         (10**7 satoshi == 0.1 BTC == 0.1 TBTC).
+    function initializeDeposit(
+        ITBTCSystem _tbtcSystem,
+        TBTCToken _tbtcToken,
+        IERC721 _tbtcDepositToken,
+        FeeRebateToken _feeRebateToken,
+        address _vendingMachineAddress,
+        uint16 _m,
+        uint16 _n,
+        uint64 _lotSizeSatoshis
+    ) public onlyFactory payable {
+        self.tbtcSystem = _tbtcSystem;
+        self.tbtcToken = _tbtcToken;
+        self.tbtcDepositToken = _tbtcDepositToken;
+        self.feeRebateToken = _feeRebateToken;
+        self.vendingMachineAddress = _vendingMachineAddress;
+        self.initialize(_m, _n, _lotSizeSatoshis);
     }
 
-    /// @notice     Notifies the contract that the courtesy period has elapsed.
-    /// @dev        This is treated as an abort, rather than fraud.
-    function notifyCourtesyTimeout() public {
-        self.notifyCourtesyTimeout();
+    /// @notice This function can only be called by the vending machine.
+    /// @dev Performs the same action as requestRedemption, but transfers
+    ///      ownership of the deposit to the specified _finalRecipient. Used as
+    ///      a utility helper for the vending machine's shortcut
+    ///      TBTC->redemption path.
+    /// @param  _outputValueBytes The 8-byte little-endian output size.
+    /// @param  _redeemerOutputScript The redeemer's length-prefixed output script.
+    /// @param  _finalRecipient     The address to receive the TDT and later be recorded as deposit redeemer.
+    function transferAndRequestRedemption(
+        bytes8 _outputValueBytes,
+        bytes memory _redeemerOutputScript,
+        address payable _finalRecipient
+    ) public {
+        require(
+            msg.sender == self.vendingMachineAddress,
+            "Only the vending machine can call transferAndRequestRedemption"
+        );
+        self.transferAndRequestRedemption(
+            _outputValueBytes,
+            _redeemerOutputScript,
+            _finalRecipient
+        );
     }
 
-    /// @notice     Withdraw caller's allowance.
-    /// @dev        Withdrawals can only happen when a contract is in an end-state.
+    /// @notice Withdraw caller's allowance.
```

Comment:
Allowance, amount, funds...

Reply:
This one was tough. I'll try a variation.